### ISSUE TITLE
Invalid reference to ForcingDataPath variable

### ISSUE THREAD

**Issue by n-beech** [esm-tools/esm_tools#809]
After updating esm-tools (6.13.9) and fesom-2.0-master and trying to run fesom2.0 standalone, the simulation doesn't start. 

The error in the compute log is:
 "forrtl: severe (19): invalid reference to variable in NAMELIST input, unit 20, file work/ab0995/a270166/runtime/fesom2/tests/2508/reinstall01/run_20010101-20010131/work/namelist.config, line 22, position 21"

Line 22 in namelist.config is: 
forcingdatapath = '/pool/data/AWICM/FESOM2/FORCING/'

Changing forcingdatapath does not solve the problem.

I get the same problem on both levante and ollie.

I would appreciate any advice on this!

**Comment by n-beech** [comment]
It seems that the problem is in the previous line in namelist.config: 
resultpath = '/work/ab0995/a270166/runtime/fesom2/tests/t3008/test10/run_20010101-20010131/work/'

Since removing the forcingdatapath variable or adding another variable after resultpath simply shifts the line and position of the error to the reference following resultpath.

**Comment by pgierz** [comment]
Hi @n-beech,

The correct way of doing this would be via namelist changes. You can see more in the handbook: https://esm-tools.readthedocs.io/en/latest/cookbook.html#changing-namelist-entries-from-the-runscript

For your concrete example, you could use in your runscript:

```yaml
fesom:
    add_namelist_changes:
        namelist.config:
            paths:
                forcingdatadir: /my/new/location         
```

I hope that helps!

**Comment by denizural** [comment]
Hi Nathan, I am also working on this issue. I will inform you today.

**Comment by n-beech** [comment]
Hi @pgierz

Unfortunately changing the forcingdatapath variable is not the problem. Regardless of what I set this variable to, I get the same error.

Thanks @denizural !

**Comment by pgierz** [comment]
Sorry, I read the initial bug report incorrectly. If deniz is working on it, then good. One hint: check that all the parameters you define in your namelist are actually also expected by fesom. It looks to me as if you are defining something fesom does not know about.

**Comment by denizural** [comment]
Hi @n-beech, the problem is solved. I ran successful simulations on both Levante and Ollie with freshly installed FESOM 2. Here are the runscripts that you need to use. I added my parts with comments:

### For Levante:
```diff
general:
    account: 'ab0995'
    setup_name: fesom
    compute_time: "00:20:00"
    initial_date: '2001-01-01'
    final_date: '2001-02-28'
    base_dir: "<your-base-dir>"
    nyear: 0
    nmonth: 1
    nday: 0
    use_venv: False

fesom:
    version: 2.0
    model_dir: "<your-model-dir>"
    pool_dir: "/pool/data/AWICM/FESOM2/"
    mesh_dir: "/pool/data/AWICM/FESOM2/MESHES/CORE2/"
    forcing_data_dir: "${pool_dir}/FORCING/"
+   # added this
+   climate_data_dir: "/pool/data/AWICM/FESOM2/INITIAL/phc3.0/"
+   # changed from res -> resolution
-   res: CORE2
+   resolution: CORE2
    lresume: False
    restart_rate: 1
    restart_first: 1
    restart_unit: 'm'
    post_processing: 0
```

### For Ollie:
```diff
general:
    setup_name: fesom
    compute_time: "00:08:00"
    initial_date: '2001-01-01'
    final_date: '2001-03-31'
    base_dir: "<your-base-dir>"
    nyear: 0
    nmonth: 1
    use_venv: False

fesom:
    version: 2.0
    model_dir: "<your-model-dir>"
    pool_dir: "/work/ollie/pool/FESOM/"
    mesh_dir: "/work/ollie/pool/FESOM/fesom2.0/meshes/mesh_CORE2_final/"
+   # changed this: res -> resolution
-   res: CORE2
+   resolution: CORE2
    lresume: 0
    restart_rate: 1
    restart_first: 1
    restart_unit: 'm'
    post_processing: 0
```

By the way, FESOM-2.1 is available. You can also use it instead. You can give these runscripts a try.

**Comment by pgierz** [comment]
Hey @denizural: as a hint for the future, you can also use the `diff` tag to a triple back-tick fence to highlight differences 🙂

**Comment by denizural** [comment]
Yes, I forgot that. Thanks for the reminder 🙂

**Comment by n-beech** [comment]
Hi @denizural, I think this may have had some side effects. 

Now running esm_master or any option with esm_master eg. -install-... returns the following error message:
No targets found for keyword fesom-2.0-esm_interface. Type 'esm_master' to get a full list
of available targets.

I am using esm_tools version 6.13.10 on levante.

**Comment by denizural** [comment]
Hi @n-beech,

I just installed Fesom 2.0 and 2.1 from scratch on Levante with esm-tool version 6.13.0. Here are the commands:
```
esm_master install-fesom-2.0
esm_master install-fesom-2.1
```
What is the command that you executed?

**Comment by n-beech** [comment]
I just reinstalled esm_tools and now it seems to be working. Thanks for your help!

**Comment by n-beech** [comment]
Hi again @denizural 

Although I thought the problem was fixed yesterday, I think you ran a successful run with the "detached head" version of fesom rather than the master branch. 

By using the command 'esm_master install-fesom-2.0' instead of 'esm_master install-fesom-2.0-master' you get two different versions of fesom with different source code.

So the problem I having with fesom-2.0-master is still an issue.

**Comment by pgierz** [comment]
Hi @n-beech,

`fesom-2.0-master` points to the bleeding edge (with the ongoing refactoring work) if your remote is [github.com/FESOM/fesom2](https://github.com/FESOM/fesom2) and is actually `fesom-2.1`, so you need different namelists. That also explains your Fortran errors: several options changed names or were removed. 

What happens if you set the version number to 2.1, as below?

```diff
fesom:
+    version: 2.1
-    version: 2.0
```

**Comment by denizural** [comment]
Hi @n-beech,

You don't get multiple versions of the model. Remember that during the Git workshop, I told that it is possible to make *time travel* with Git. This is what happens when you run esm_master:

| command | Git action |
| --- | --- |
| esm_master install-fesom-2.0 | `git clone -b 2.0.2 https://github.com/FESOM/fesom2.git fesom-2.0` |
| esm_master install-fesom-2.1 | `git clone -b 2.1.0 https://github.com/FESOM/fesom2.git fesom-2.1` |
| esm_master install-fesom-2.0-master | `git clone -b refactoring https://github.com/FESOM/fesom2.git fesom-2.0` |

Basically, you download the fesom-2 repository and checkout to the tag. Remember that tag basically marks a point in the development history. Since it is not the most recent commit in the branch, you get the *detached head* warning from Git. This is not an error but simply tells you that if you make any commits, they won't belong to a branch.

Is this helpful for you?

**Comment by n-beech** [comment]
Thanks I see the problem now. 

@pgierz changing version in the namelist to 2.1 did not remove the error but if the master branch essentially points to fesom-2.1, I will just install and use fesom-2.1.

@denizural Sorry I wasn't in the git workshop. But I understand the difference here, thanks.

**Comment by denizural** [comment]
Hi @n-beech, 
sorry my bad 🤦‍♂️.  Now, I remember that you were in the ESM-Tools but not in the Git one. You can try it with Fesom 2.1. It is the latest version of Fesom. Just make sure that your model directory does not have the Fesom directory with the same name. Eg. if `Fesom-2.1/` already exists running esm-master to install `Fesom-2.1` may crash because it does not overwrite your directory.

**Comment by n-beech** [comment]
Yes, sure, but installing fesom-2.0-master creates a directory with the name fesom-2.0. 

So if I understand correctly, installing fesom-2.1 shouldn't cause a crash, even if the fesom-2.0 directory contains essentially the same model.

**Comment by denizural** [comment]
Yes, in this command `git clone -b 2.0.2 https://github.com/FESOM/fesom2.git fesom-2.0` the last argument is the name of the directory to be created. So by default Git would create `fesom2` but we explicitly tell it to create `fesom-2.0`. For Fesom-2.1 this will be `fesom-2.1` as you can see from the post I made above: https://github.com/esm-tools/esm_tools/issues/809#issuecomment-1235239782
